core/src/setup/Kokkos_Setup_SYCL.hpp

```diff
@@ -45,4 +45,21 @@
 #define KOKKOS_IMPL_SYCL_GET_MULTI_PTR(accessor) accessor.get_pointer()
 #endif
 
+// FIXME_SYCL Use type directly once it has stabilized in SYCL.
+namespace Kokkos::Impl::SYCLTypes {
+#ifndef SYCL_EXT_INTEL_USM_ADDRESS_SPACES
+#error SYCL_EXT_INTEL_USM_ADDRESS_SPACES undefined!
+#elif SYCL_EXT_INTEL_USM_ADDRESS_SPACES >= 2
+template <typename T>
+using device_ptr = sycl::ext::intel::device_ptr<T>;
```

Comment:
I would prefer if didn't introduced that nested typename and renamed 
`Kokkos::Impl::SYCLTypes::{host,device}_ptr` to `Kokkos::Impl::sycl_{host,device}_ptr` which I find more readable and consistent with common practice throughout the codebase.

Reply:
It also makes most if not all of the changes in this PR be simply
`sycl::device_ptr -> sycl_device_ptr` and so on.

Reply:
I don't have a strong opinion. If that gets me your approval, I'm happy to change it.

Reply:
See 6333d7d